include/aws/auth/credentials.h

```diff
@@ -216,6 +216,11 @@ struct aws_credentials_provider_imds_options {
      */
     enum aws_imds_protocol_version imds_version;
 
+    /*
+     * If true, fallback from v2 to v1 will be disabled for all cases
+     */
+    bool ec2_metadata_v1_disabled;
```

Comment:
`ec2_metadata_v1_disabled`, but you can still set the `imds_version` to V1 manually.

And, it's already part of IMDS options, I think we can skip the `ec2_metadata` part

Maybe `v1_fallback_disabled`?

Reply:
`ec2_metadata_v1_disabled` will be used cross SDKs along with the relevant env variable.